workspaces/src/types/mod.rs

```diff
@@ -72,13 +105,82 @@ impl From<PublicKey> for near_crypto::PublicKey {
 
 /// Public key of an account on chain. Usually created along with a [`SecretKey`]
 /// to form a keypair associated to the account.
-#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[derive(
+    Debug,
+    Clone,
+    Hash,
+    Eq,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    Serialize,
+    Deserialize,
+    BorshSerialize,
+    BorshDeserialize,
+)]
 pub struct PublicKey(pub(crate) near_crypto::PublicKey);
 
+#[allow(clippy::len_without_is_empty)] // PublicKey is guaranteed to never be empty due to KeyType restrictions.
 impl PublicKey {
+    /// Create an empty `PublicKey` with the given [`KeyType`]. This is a zero-ed out public key with the
+    /// length of the bytes determined by the associated key type.
+    pub fn empty(key_type: KeyType) -> Self {
+        Self(near_crypto::PublicKey::empty(key_type.into_near_keytype()))
+    }
+
+    /// Get the number of bytes this key uses. This will differ depending on the [`KeyType`]. i.e. for
+    /// ED25519 keys, this will return 32 + 1, while for SECP256K1 keys, this will return 64 + 1. The +1
+    /// is used to store the key type, and will appear at the start of the serialized key.
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    /// Get the [`KeyType`] of the public key.
     pub fn key_type(&self) -> KeyType {
         KeyType::from_near_keytype(self.0.key_type())
     }
+
+    /// Get the key data of the public key. This is serialized bytes of the public key. This will not
+    /// include the key type, and will only contain the raw key data.
+    pub fn key_data(&self) -> &[u8] {
+        self.0.key_data()
+    }
+}
+
+impl Display for PublicKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl FromStr for PublicKey {
+    type Err = Error;
+
+    fn from_str(value: &str) -> Result<Self, Self::Err> {
+        let pk = near_crypto::PublicKey::from_str(value)
+            .map_err(|e| ErrorKind::DataConversion.custom(e))?;
+
+        Ok(Self(pk))
+    }
+}
+
+impl From<PublicKey> for Vec<u8> {
+    fn from(pk: PublicKey) -> Vec<u8> {
+        let mut data = vec![pk.key_type() as u8];
+        data.extend(pk.key_data());
+        data
+    }
+}
```

Comment:
This sort of feels like an antipattern, let's maybe just leave it to borsh?

Reply:
you have a good point there 👍 . Originally ported this over from SDK, but the borsh call is much more explicit anyways since it's called `try_to_vec` instead of purely just `into`